electrum/commands.py

```diff
@@ -1474,17 +1476,20 @@ async def check_hold_invoice(self, payment_hash: str, wallet: Abstract_Wallet =
         status = "unknown"
         if info is None:
             pass
-        elif not is_accepted_mpp:
+        elif not is_accepted_mpp and not wallet.lnworker.get_preimage_hex(payment_hash):
+            # is_accepted_mpp is False for settled payments
             status = "unpaid"
         elif is_accepted_mpp and payment_hash in wallet.lnworker.dont_settle_htlcs:
             status = "paid"
-        elif (payment_hash in wallet.lnworker._preimages
-                and payment_hash not in wallet.lnworker.dont_settle_htlcs
-                and is_accepted_mpp):
+        elif wallet.lnworker.get_preimage_hex(payment_hash) is not None \
+                and payment_hash not in wallet.lnworker.dont_settle_htlcs:
             status = "settled"
+            plist = wallet.lnworker.get_payments(status='settled')[bfh(payment_hash)]
+            _dir, amount_msat, _fee, _ts = wallet.lnworker.get_payment_value(info, plist)
+            amount_sat = amount_msat // 1000
         result = {
             "status": status,
-            "amount_sat": amount_sat
+            "amount_sat_received": amount_sat
         }
```

Comment:
Perhaps we could add another field `invoice_amount_sat` or similar. It might be useful as a sanity check for the user.
Also, nit, `_sat` I think is nicer as a suffix, so perhaps `received_amount_sat` is better name?

```suggestion
            "received_amount_sat": amount_sat,
        }
        if info is not None:
            result["invoice_amount_sat"] = (info.amount_msat or 0) // 1000
```

Reply:
Ok yes could be useful, committed that in https://github.com/spesmilo/electrum/pull/10059/commits/16785fed976fd96446852fa8819100b5dd4f0ab5